### ISSUE TITLE
pip install fails due to htslib

### ISSUE THREAD

**Issue by vivekiitkgp** [pysam-developers/pysam#281]
Hi,

I'm trying to install `pysam` on a Mac OS X 10.5 but the install is failing due to an error with `htslib`. Here is a log of the output.

```
(py2)❯ pip install pysam
Collecting pysam
  Using cached pysam-0.9.0.tar.gz
Building wheels for collected packages: pysam
  Running setup.py bdist_wheel for pysam ... error
  Complete output from command /Users/vrai/.anaconda/envs/py2/bin/python -u -c "import setuptools, tokenize;__file__='/private/var/folders/f6/06x0cvfd52q04cjg70b8w4xxrr0vm1/T/pip-build-ImDmvE/pysam/setup.py';exec(compile(getattr(tokenize, 'open', open)(__file__).read().replace('\r\n', '\n'), __file__, 'exec'))" bdist_wheel -d /var/folders/f6/06x0cvfd52q04cjg70b8w4xxrr0vm1/T/tmpW71wS0pip-wheel- --python-tag cp27:
  # pysam: htslib mode is separate
  checking for gcc... gcc
  checking whether the C compiler works... yes
  checking for C compiler default output file name... a.out
  checking for suffix of executables...
  checking whether we are cross compiling... no
  checking for suffix of object files... o
  checking whether we are using the GNU C compiler... yes
  checking whether gcc accepts -g... yes
  checking for gcc option to accept ISO C89... none needed
  checking for ranlib... ranlib
  checking how to run the C preprocessor... gcc -E
  checking for grep that handles long lines and -e... /usr/bin/grep
  checking for egrep... /usr/bin/grep -E
  checking for ANSI C header files... yes
  checking for sys/types.h... yes
  checking for sys/stat.h... yes
  checking for stdlib.h... yes
  checking for string.h... yes
  checking for memory.h... yes
  checking for strings.h... yes
  checking for inttypes.h... yes
  checking for stdint.h... yes
  checking for unistd.h... yes
  checking for stdlib.h... (cached) yes
  checking for unistd.h... (cached) yes
  checking for sys/param.h... yes
  checking for getpagesize... yes
  checking for working mmap... yes
  checking for gmtime_r... yes
  checking for zlib.h... yes
  checking for inflate in -lz... yes
  checking for curl_easy_pause in -lcurl... yes
  checking for CCHmac... yes
  configure: creating ./config.status
  config.status: creating config.mk
  config.status: creating config.h
  config.status: config.h is unchanged
  # pysam: htslib configure options: --enable-libcurl
  # pysam: libcurl of builtin htslib has been enabled, adding shared libcurl and libcrypto
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.5-x86_64-2.7
  creating build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/__init__.py -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/bcftools.py -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/config.py -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/namedtuple.py -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/Pileup.py -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/samtools.py -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/utils.py -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/version.py -> build/lib.macosx-10.5-x86_64-2.7/pysam
  creating build/lib.macosx-10.5-x86_64-2.7/pysam/include
  copying pysam/include/__init__.py -> build/lib.macosx-10.5-x86_64-2.7/pysam/include
  package init file 'htslib/__init__.py' not found (or not a regular file)
  package init file 'htslib/htslib/__init__.py' not found (or not a regular file)
  package init file 'samtools/__init__.py' not found (or not a regular file)
  package init file 'bcftools/__init__.py' not found (or not a regular file)
  package init file 'samtools/win32/__init__.py' not found (or not a regular file)
  copying pysam/calignedsegment.pxd -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/calignmentfile.pxd -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/cbcf.pxd -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/cfaidx.pxd -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/chtslib.pxd -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/csamfile.pxd -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/ctabix.pxd -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/ctabixproxies.pxd -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/cutils.pxd -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/cvcf.pxd -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/htslib_util.h -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/pysam_stream.h -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/pysam_util.h -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/samfile_util.h -> build/lib.macosx-10.5-x86_64-2.7/pysam
  copying pysam/tabix_util.h -> build/lib.macosx-10.5-x86_64-2.7/pysam
  creating build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib
  copying htslib/config.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib
  copying htslib/hfile_internal.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib
  copying htslib/hts_internal.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib
  copying htslib/version.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib
  creating build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/bgzf.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/cram.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/faidx.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/hfile.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/hts.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/hts_defs.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/kbitset.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/kfunc.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/khash.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/khash_str2int.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/klist.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/knetfile.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/kseq.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/ksort.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/kstring.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/regidx.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/sam.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/synced_bcf_reader.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/tbx.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/vcf.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/vcf_sweep.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  copying htslib/htslib/vcfutils.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/htslib/htslib
  creating build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/bam.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/bam2bcf.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/bam_endian.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/bam_lpileup.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/bam_plbuf.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/bam_tview.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/errmod.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/kprobaln.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/pysam.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/sam.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/sam_header.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/sam_opts.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/sample.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/samtools.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/stats_isize.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  copying samtools/version.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools
  creating build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/bcftools.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/call.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/convert.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/filter.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/gvcf.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/HMM.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/khash_str2str.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/kmin.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/ploidy.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/prob1.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/pysam.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/rbuf.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/tsv2vcf.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/vcmp.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  copying bcftools/version.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/bcftools
  creating build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools/win32
  copying samtools/win32/xcurses.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools/win32
  copying samtools/win32/zconf.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools/win32
  copying samtools/win32/zlib.h -> build/lib.macosx-10.5-x86_64-2.7/pysam/include/samtools/win32
  running build_ext
  building 'pysam.libchtslib' extension
  creating build/temp.macosx-10.5-x86_64-2.7
  creating build/temp.macosx-10.5-x86_64-2.7/pysam
  creating build/temp.macosx-10.5-x86_64-2.7/htslib
  creating build/temp.macosx-10.5-x86_64-2.7/htslib/cram
  gcc -fno-strict-aliasing -I/Users/vrai/.anaconda/envs/py2/include -arch x86_64 -isysroot /Developer/SDKs/MacOSX10.5.sdk -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Ipysam -I. -Ihtslib -I/Users/vrai/.anaconda/envs/py2/include/python2.7 -c pysam/chtslib.c -o build/temp.macosx-10.5-x86_64-2.7/pysam/chtslib.o -Wno-unused -Wno-strict-prototypes -Wno-sign-compare -Wno-error=declaration-after-statement
  gcc -fno-strict-aliasing -I/Users/vrai/.anaconda/envs/py2/include -arch x86_64 -isysroot /Developer/SDKs/MacOSX10.5.sdk -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Ipysam -I. -Ihtslib -I/Users/vrai/.anaconda/envs/py2/include/python2.7 -c pysam/htslib_util.c -o build/temp.macosx-10.5-x86_64-2.7/pysam/htslib_util.o -Wno-unused -Wno-strict-prototypes -Wno-sign-compare -Wno-error=declaration-after-statement
  gcc -fno-strict-aliasing -I/Users/vrai/.anaconda/envs/py2/include -arch x86_64 -isysroot /Developer/SDKs/MacOSX10.5.sdk -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Ipysam -I. -Ihtslib -I/Users/vrai/.anaconda/envs/py2/include/python2.7 -c htslib/bgzf.c -o build/temp.macosx-10.5-x86_64-2.7/htslib/bgzf.o -Wno-unused -Wno-strict-prototypes -Wno-sign-compare -Wno-error=declaration-after-statement
  gcc -fno-strict-aliasing -I/Users/vrai/.anaconda/envs/py2/include -arch x86_64 -isysroot /Developer/SDKs/MacOSX10.5.sdk -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Ipysam -I. -Ihtslib -I/Users/vrai/.anaconda/envs/py2/include/python2.7 -c htslib/faidx.c -o build/temp.macosx-10.5-x86_64-2.7/htslib/faidx.o -Wno-unused -Wno-strict-prototypes -Wno-sign-compare -Wno-error=declaration-after-statement
  gcc -fno-strict-aliasing -I/Users/vrai/.anaconda/envs/py2/include -arch x86_64 -isysroot /Developer/SDKs/MacOSX10.5.sdk -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Ipysam -I. -Ihtslib -I/Users/vrai/.anaconda/envs/py2/include/python2.7 -c htslib/hfile.c -o build/temp.macosx-10.5-x86_64-2.7/htslib/hfile.o -Wno-unused -Wno-strict-prototypes -Wno-sign-compare -Wno-error=declaration-after-statement
  gcc -fno-strict-aliasing -I/Users/vrai/.anaconda/envs/py2/include -arch x86_64 -isysroot /Developer/SDKs/MacOSX10.5.sdk -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -Ipysam -I. -Ihtslib -I/Users/vrai/.anaconda/envs/py2/include/python2.7 -c htslib/hfile_libcurl.c -o build/temp.macosx-10.5-x86_64-2.7/htslib/hfile_libcurl.o -Wno-unused -Wno-strict-prototypes -Wno-sign-compare -Wno-error=declaration-after-statement
  htslib/hfile_libcurl.c:93:10: error: use of undeclared identifier 'CURLE_NOT_BUILT_IN'
      case CURLE_NOT_BUILT_IN:
           ^
  htslib/hfile_libcurl.c:109:10: error: use of undeclared identifier 'CURLE_REMOTE_ACCESS_DENIED'; did you mean 'CURLE_FTP_ACCESS_DENIED'?
      case CURLE_REMOTE_ACCESS_DENIED:
           ^~~~~~~~~~~~~~~~~~~~~~~~~~
           CURLE_FTP_ACCESS_DENIED
  /Developer/SDKs/MacOSX10.5.sdk/usr/include/curl/curl.h:316:3: note: 'CURLE_FTP_ACCESS_DENIED' declared here
    CURLE_FTP_ACCESS_DENIED,       /* 9 a service was denied by the FTP server
    ^
  htslib/hfile_libcurl.c:125:16: error: use of undeclared identifier 'ENOTRECOVERABLE'
          return ENOTRECOVERABLE; // Indicates bugs in our callback routines
                 ^
  htslib/hfile_libcurl.c:133:10: error: use of undeclared identifier 'CURLE_RANGE_ERROR'
      case CURLE_RANGE_ERROR:
           ^
  htslib/hfile_libcurl.c:150:10: error: use of undeclared identifier 'CURLE_REMOTE_DISK_FULL'; did you mean 'CURLE_TFTP_DISKFULL'?
      case CURLE_REMOTE_DISK_FULL:
           ^~~~~~~~~~~~~~~~~~~~~~
           CURLE_TFTP_DISKFULL
  /Developer/SDKs/MacOSX10.5.sdk/usr/include/curl/curl.h:386:3: note: 'CURLE_TFTP_DISKFULL' declared here
    CURLE_TFTP_DISKFULL,           /* 70 - out of disk space on server */
    ^
  htslib/hfile_libcurl.c:153:10: error: use of undeclared identifier 'CURLE_REMOTE_FILE_EXISTS'
      case CURLE_REMOTE_FILE_EXISTS:
           ^
  htslib/hfile_libcurl.c:265:54: error: use of undeclared identifier 'CURL_WRITEFUNC_PAUSE'
      if (n > fp->buffer.len) { fp->paused = 1; return CURL_WRITEFUNC_PAUSE; }
                                                       ^
  htslib/hfile_libcurl.c:283:11: warning: implicit declaration of function 'curl_easy_pause' is invalid in C99 [-Wimplicit-function-declaration]
      err = curl_easy_pause(fp->easy, CURLPAUSE_CONT);
            ^
  htslib/hfile_libcurl.c:283:37: error: use of undeclared identifier 'CURLPAUSE_CONT'
      err = curl_easy_pause(fp->easy, CURLPAUSE_CONT);
                                      ^
  htslib/hfile_libcurl.c:309:39: error: use of undeclared identifier 'CURL_READFUNC_PAUSE'
          else { fp->paused = 1; return CURL_READFUNC_PAUSE; }
                                        ^
  htslib/hfile_libcurl.c:328:37: error: use of undeclared identifier 'CURLPAUSE_CONT'
      err = curl_easy_pause(fp->easy, CURLPAUSE_CONT);
                                      ^
  htslib/hfile_libcurl.c:398:37: error: use of undeclared identifier 'CURLPAUSE_CONT'
      err = curl_easy_pause(fp->easy, CURLPAUSE_CONT);
                                      ^
  htslib/hfile_libcurl.c:425:37: error: use of undeclared identifier 'CURLPAUSE_CONT'
      err = curl_easy_pause(fp->easy, CURLPAUSE_CONT);
                                      ^
  1 warning and 12 errors generated.
  error: command 'gcc' failed with exit status 1

  ----------------------------------------
  Failed building wheel for pysam
  Running setup.py clean for pysam
Failed to build pysam
```

I suspect the line `libcurl of builtin htslib has been enabled` to be the cause and disabling it should work. Though not tried yet.

**Comment by serverhorror** [comment]
I'm running into the same problem. This currently prevents RSeQC from building since it pulls in (latest) pysam via dependencies. I had to manually `pip install 'pysam<0.9'` and then `pip install rseqc`.

* OS: Scientific Linux 6.6
* Python 2.7.9 built from source.

**Comment by mickey2016** [comment]
I encountered the same error on 3 CentOS, and serverhorror's comment solved the problem, thanks!
CentOS release 6.6 (Final)
CentOS release 6.4 (Final)
CentOS release 6.2 (Final)

**Comment by JDiprose** [comment]
Not sure about the Mac issue, but the RHEL6 issues probably relate to:
https://github.com/samtools/htslib/commit/2805ae8041f2d49b889d0b4f54c53070e4791dbb
Applying the fix allowed it to build for me.

**Comment by AndreasHeger** [comment]
Thanks, this should be resolved with htslib 1.3.1 and pysam 0.9.1
